### PR TITLE
Document notifications technical capabilities

### PR DESCRIPTION
**Pull request by thevoiceofzeke**
[MUMUP-2856](https://jira.doit.wisc.edu/jira/browse/MUMUP-2856)

[MUMUP-2857](https://jira.doit.wisc.edu/jira/browse/MUMUP-2857)

**In this PR:**
- Wrote documentation for notifications, including user-facing content and technical implementation
- Removed the `action` attribute from `actionButtons` configuration (limited use case and not currently working as intended)
- Included minimal best practices guidance for language, length, dismissability, and priority flagging.

### Screenshots
screen shot 2017-02-27 at 11 42 18 am
screen shot 2017-02-27 at 11 42 23 am

### CODE REVIEW

File: docs/notifications.md [hunk 1]
```diff
@@ -1,2 +1,106 @@
...
+### Mobile menu link
+
+In addition to the bell icon, the mobile menu contains a link to the notifications page that also displays the number of unseen notifications.
+
+[![mobile menu notifications link](./img/notifications/mobile-link.png)](img/notifications/mobile-link.png)
+
+### Priority notifications
+
+For critical notifications that require a higher degree of visibility, notifications can be designated as "priority" and will appear more prominently, fixed above the
+application top bar. In the case that the user has multiple unseen priority notifications, a generic message will be displayed, featuring the priority notifications
```
Comment (apetro) [inline docs/notifications.md]: I don't think the word "unseen" here is quite right.

File: docs/notifications.md [hunk 2]
```diff
@@ -1,2 +1,106 @@
...
+On the notifications page, users can view their notifications, dismiss them, and follow calls to action. They can also click the
+'Dismissed' tab to view notifications they've previously dismissed.
+
+Priority notification float to the top of the list and appear with a small icon to the left of the notification's content.
+
+[![notifications page](./img/notifications/notifications-page.png)](img/notifications/notifications-page.png)
+
+## Technical implementation
+
+Notifications depend on a JSON feed containing certain attributes and flags. See the [configuraton]() doc for information about how to
```
Comment (apetro) [inline docs/notifications.md]: The `[configuraton]()` link syntax looks bugged.

File: docs/notifications.md [hunk 3]
```diff
@@ -1,2 +1,106 @@
...
+      "dismissable" : true,
+      "priority" : false
+    }
+  ]
+}
+```
+
+**Attribute breakdown**
+
+- **id**: A unique number to identify the notification. This is used for sort order on the notifications page.
```
Comment (apetro) [inline docs/notifications.md]: Also keys dismissed-ness.

File: docs/notifications.md [hunk 4]
```diff
@@ -1,2 +1,106 @@
...
+- **groups**: An attribute to optionally show notifications only to specific groups (ex. Manifest groups, uPortal groups). **Must contain at least one value**. Using the "Everyone" group will make
+your notification visible to all users. Contact your portal development team for more information about group filtering.
+- **title**: The text to be displayed as the notification's main content. **Be concise!** Try to limit your notification's title to ~140 characters. Longer titles are less likely
+to be read and can cause minor display errors.
+- **actionURL**: A URL where users can get more information or respond to calls to action.
+- **actionAlt** (*optional*): Applies an aria-label to the notification title. Use this if vision-impaired users might need additional context to understand your notification.
+- **dismissable**: Set to true if users should be able to dismiss the notification from their list. This also works for dismissing priority notifications from their fixed position above the top bar.
+**This should almost always be true**.
```
Comment (apetro) [inline docs/notifications.md]: This should be true when notifications are not self-dismissing, that is, when users do not drop out of their audience groups when they address whatever they're being notified about.

File: docs/notifications.md (hunk 1 above)
Comment (thevoiceofzeke) [inline docs/notifications.md]: Adjusted

File: docs/notifications.md (hunk 4 above)
Comment (thevoiceofzeke) [inline docs/notifications.md]: Ah, so the cases when taking the action would remove them from the group filtering for that notification (ex. activating their NetID)? I can add a bit about that, but I'm wondering more about when it should be set to false, if ever.

File: docs/notifications.md (hunk 3 above)
Comment (thevoiceofzeke) [inline docs/notifications.md]: Thanks!

File: docs/notifications.md (hunk 2 above)
Comment (thevoiceofzeke) [inline docs/notifications.md]: I just forgot to include the link :P

**Pull request merged**